libminifi/src/Funnel.cpp

```diff
@@ -0,0 +1,26 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Funnel.h"
+
+namespace org::apache::nifi::minifi {
+
+void Funnel::onSchedule(core::ProcessContext&, core::ProcessSessionFactory&) {
+  addAutoTerminatedRelationship(Success);
```

Comment:
isn't it a problem that it's always auto-terminated, even when connected to something else?

Reply:
If there is a real connection, that will take precedence. Auto-termination is only checked if the output is not connected to anything.